### PR TITLE
[43842] Re-design and small changes to modal dialogues (DS) for coherence

### PR DESCRIPTION
**Pull request by bsatarnejad**
Modal redesign changes:

- [x] WP deletion changes:

The header section does not have a divider.
The text has been modified to include the name of the WP inside.
The action button changed to “Delete” and now includes an icon.

- [x] Invite user:

Modal dialog doesn’t include “X” button on the header anymore.
The action bar now contains the option button “Back” and the always present option to “cancel”
The process has been simplified to only three steps instead of one step per each form field.
Now the form fields “Name or email address”, “Role in project [X]” and “Invitation message” are in a single step.
The guidelines texts in the “Role in project [X]” and “Invitation message” are now more generic (no user name).
The last step now also includes the text from the field “Invitation message”.

- [x] Log time:

Modal dialog doesn’t include “X” button on the header anymore.

- [x] Add widget:

Modal dialog doesn’t include “X” button on the header anymore.
The list in the middle doesn’t have dividers anymore as now is using the spot-list default component as base.
There is an action bar with only one action “Cancel”.

- [x] New Board:

Modal dialog doesn’t include “X” button on the header anymore.
There is an action bar with only one action “Cancel”.
The new width is 60REM (960px) as 70REM is removed as width size.

- [x] Board > Configure this board:

Modal dialog doesn’t include “X” button on the header anymore.
The size of the modal depends on the content and not a fix height.

- [x] Work package table > Table settings: 

Modal dialogs doesn’t include “X” button on the header anymore.
The height of the modal is now fixed depending on the highest tab option (in this case the tab “Display settings” in order to avoid flickering behavior. 
The action bar is now gray and the order of the buttons is with the positive action in the right. 
The new width is 60REM (960px) as 70REM is removed as width size.

- [x] Work package table > Export:

Modal dialogs doesn’t include “X” button on the header anymore.
There is an action bar with only one action “Cancel”.

- [x] Work package table/Calendar/Board/Team planner > Save as:

Modal dialog doesn’t include “X” button on the header anymore.
The divider in the content has been switched with the “Visibility settings” header.
“Visibility settings” header font is reduced.
The action bar now includes two buttons “Cancel” and “Save”. Only the ”Save” includes icon.

- [x] Work package table/Calendar/Board/Team planner > Visibility settings:

Modal dialogs doesn’t include “X” button on the header anymore.
The size of the modal depends on the content and not a fix height.
The action bar now includes two buttons “Cancel” and “Save”. Only the ”Save” includes icon.

- [x] Help text:

Modal dialogs doesn’t include “X” button on the header anymore.
The header section does not have divider.
The action bar now includes two buttons “Close” and “Edit”.

- [x] Help > Introduction video:

Modal dialogs doesn’t include “X” button on the header anymore.
The header section does not have color.
The action bar now includes two buttons“Close”.
The text ”You can restart this video from the help menu” is now part of the description.

- [x] My account > Two-factor authentication > Backup codes password:

Modal dialogs doesn’t include “X” button on the header anymore.
The action bar now includes two buttons“Cancel” and “Confirm”.

- [x] My account > Two-factor authentication > Delete authentication:

Modal dialogs doesn’t include “X” button on the header anymore. 
The action bar now includes two buttons“Cancel” and “Confirm”.

- [x] Administration > Attribute help texts > Preview help text:

Modal dialogs doesn’t include “X” button on the header anymore.
The header section does not have divider. 
The action bar now includes two buttons “Close” and “Edit”.

- [x] Administration > Enterprise Edition > Delete token:

Modal dialogs doesn’t include “X” button on the header anymore.
The header section does not have divider.
All texts has been actualized to make the action clearer.
The action bar now includes two buttons “Cancel” and “Delete”. The second one now includes icon.

- [x] Administration > GDPR > Review DPA: 

Modal dialogs doesn’t include “X” button on the header anymore.
Now there is an action bar with the button “Close”.

- [x] Administration > API and webhooks > webhooks

Change header and body specified font size

https://community.openproject.org/projects/openproject/work_packages/43842/activity

### CODE REVIEW

**Comment by Kharonus** [comment]
Should we already review this or wait, until all checkboxes are checked?

**Comment by bsatarnejad** [comment]
> Should we already review this or wait, until all checkboxes are checked?

Hi @Kharonus , I am still working on the PR. Some of the changes are applied.

File: frontend/src/app/shared/components/modals/wp-destroy-modal/wp-destroy.modal.html [hunk 1]
```diff
@@ -1,22 +1,20 @@
 <div
-  class="spot-modal wp-table--configuration-modal loading-indicator--location"
+  class="spot-modal loading-indicator--location"
```
Comment (b12f) [inline frontend/src/app/shared/components/modals/wp-destroy-modal/wp-destroy.modal.html]: Does this even have a loading state? I think we can delete it

File: frontend/src/app/shared/components/modals/wp-destroy-modal/wp-destroy.modal.html [hunk 2]
```diff
@@ -74,10 +72,11 @@
       ></button>
       <button
         class="button button_no-margin -danger spot-action-bar--action"
-        [textContent]="text.confirm"
         [attr.disabled]="busy || blockedDueToUnconfirmedChildren || undefined"
-        (click)="confirmDeletion($event)"
-      ></button>
+        (click)="confirmDeletion($event)">
+        <op-icon icon-classes="button--icon icon-delete"></op-icon>
```
Comment (b12f) [inline frontend/src/app/shared/components/modals/wp-destroy-modal/wp-destroy.modal.html]: Icons can now just be a span:
```suggestion
        <span class="spot-icon spot-icon_delete"></span>
```

Comment (b12f) [reply frontend/src/app/shared/components/modals/wp-destroy-modal/wp-destroy.modal.html]: might still need the `button--icon` class tho

File: frontend/src/app/features/invite-user-modal/invite-user.component.html [hunk 3]
```diff
@@ -9,42 +9,29 @@
...
-<op-ium-message
-  *ngIf="!loading && step === Steps.Message"
-  class="spot-modal"
-  [message]="message"
-  [principal]="principalData.principal"
-  [project]="project"
-  (save)="onMessageSave($event)"
-  (back)="goTo(Steps.Role)"
-  (close)="closeMe()"
-></op-ium-message>
```
Comment (b12f) [inline frontend/src/app/features/invite-user-modal/invite-user.component.html:46]: The component files for this and for the role step still need to be deleted

File: frontend/src/app/features/invite-user-modal/principal/principal.component.ts [hunk 4]
```diff
@@ -54,31 +62,50 @@ export class PrincipalComponent implements OnInit {
...
+      description: ():string => this.I18n.t('js.invite_user_modal.role.description', {
+        principal: this.principal?.name,
+      }),
+      required: this.I18n.t('js.invite_user_modal.role.required'),
+    },
+    message: {
+      label: this.I18n.t('js.invite_user_modal.message.label'),
+      description: ():string => this.I18n.t('js.invite_user_modal.message.description', {
+        principal: this.principal?.name,
+      }),
```
Comment (b12f) [inline frontend/src/app/features/invite-user-modal/principal/principal.component.ts:98]: The help texts for role and message are currently broken as long as a principal has not been selected yet:


20220919_13h40m34s_grim

File: frontend/src/app/features/invite-user-modal/principal/principal.component.html [hunk 5]
```diff
@@ -60,16 +60,69 @@
...
+        [innerHtml]="text.role.description()"
+      ></p>
+    
+      <div
+        slot="errors"
+        class="op-form-field--error"
+        *ngIf="roleControl?.touched && roleControl?.invalid"
+        >
+          {{ text.role.required }}
+        </div>
```
Comment (b12f) [inline frontend/src/app/features/invite-user-modal/principal/principal.component.html:84]: Not sure why, but the errors for role and principal are not shown correctly. To test, just open the second page and press next without changing anything:

20220919_13h42m22s_grim

File: frontend/src/app/shared/components/grids/widgets/add/add.modal.html [hunk 6]
```diff
@@ -25,4 +25,16 @@
       class="grid--addable-widget"
     ></div>
   </div>
+
```
Comment (b12f) [inline frontend/src/app/shared/components/grids/widgets/add/add.modal.html:39]: The list above here should be a `spot-list` but it not yet

File: frontend/src/app/features/work-packages/components/wp-table/configuration-modal/wp-table-configuration.modal.html [hunk 7]
```diff
@@ -24,19 +24,19 @@
   </div>
```
Comment (b12f) [inline frontend/src/app/features/work-packages/components/wp-table/configuration-modal/wp-table-configuration.modal.html:24]: In this modal, the whole modal scrolls instead of the tab contents. I would expect this to be the other way around:


20220919_13h51m54s_grim

File: frontend/src/app/shared/components/modals/save-modal/save-query.modal.html [hunk 8]
```diff
@@ -30,7 +30,8 @@
     </div>
 
     <section class="form--section">
-      <h3 [textContent]="text.label_visibility_settings"></h3>
+      <div class="spot-divider"></div>
+      <p>{{ text.label_visibility_settings }}</p>
```
Comment (b12f) [inline frontend/src/app/shared/components/modals/save-modal/save-query.modal.html]: This has lost all heading styles. h3 might have been much, but making it a paragraph makes it unclear that this is supposed to be a heading:

20220919_13h53m48s_grim

File: app/views/onboarding/_onboarding_video_modal.html.erb [hunk 9]
```diff
@@ -46,9 +46,8 @@ See COPYRIGHT and LICENSE files for more details.
```
Comment (b12f) [inline app/views/onboarding/_onboarding_video_modal.html.erb:46]: Honestly I also think we can remove the icon from this modal header, but maybe @marcalcobe and @psatyal have different opinions?

20220919_13h59m37s_grim

**Review by b12f — CHANGES_REQUESTED** [review]
A ton of great work, but there's some last "Feinschliff" to be done :)

File: frontend/src/app/shared/components/modals/wp-destroy-modal/wp-destroy.modal.html (hunk 2 above)
Comment (Kharonus) [reply frontend/src/app/shared/components/modals/wp-destroy-modal/wp-destroy.modal.html]: I added some overwrite rules to use simple spans of `spot-icon` within the "old" buttons, but this is not merged yet.
See [here](https://github.com/opf/openproject/commit/12ed963f5fca1f3196edc19f456ab6f81680e486#diff-2c53795f153acc4b1f50994a3333ddaf2f6ef82a8ceee970352cc74730af1fb9).

But without this kind of overwrites `spot-icon` spans does not work within the old buttons.

File: frontend/src/app/features/invite-user-modal/principal/principal.component.ts (hunk 4 above)
Comment (bsatarnejad) [reply frontend/src/app/features/invite-user-modal/principal/principal.component.ts:98]: I made changes in en.yml file for the messages, maybe it will take more time or after merging, the translation for other languages like German will be updated by Crowdin. Is it correct? or Should I change them myself?

Comment (b12f) [reply frontend/src/app/features/invite-user-modal/principal/principal.component.ts:98]: Ah I see, you're totally right, I'm sorry!

File: frontend/src/app/features/invite-user-modal/principal/principal.component.ts [hunk 10]
```diff
@@ -89,38 +116,54 @@ export class PrincipalComponent implements OnInit {
...
   }
 
-  get dynamicFieldsControl() {
+  get dynamicFieldsControl():AbstractControl|null {
     return this.principalForm.get('userDynamicFields');
   }
 
   get customFields():{ [key:string]:any } {
-    return this.dynamicFieldsControl?.value;
+    return this.dynamicFieldsControl?.value as { [key:string]:any };
```
Comment (Kharonus) [inline frontend/src/app/features/invite-user-modal/principal/principal.component.ts]: Just a minor hint: if you revert this change, line 148 will still has a linter error, but your PR would be green, as you didn't changed this line.

File: app/views/onboarding/_onboarding_video_modal.html.erb (hunk 9 above)
Comment (bsatarnejad) [reply app/views/onboarding/_onboarding_video_modal.html.erb:46]: As we discussed it in a meeting with designers, we can keep the icon here.

File: config/locales/js-en.yml [hunk 11]
```diff
@@ -1131,12 +1131,14 @@ en:
...
       button_submit: "Submit"
       button_cancel: "Cancel"
+      button_delete: "Delete"
       form_submit:
         title: 'Confirm to continue'
         text: 'Are you sure you want to perform this action?'
       destroy_work_package:
         title: "Confirm deletion of %{label}"
-        text: "Are you sure you want to delete the following %{label} ?"
+        single_text: "Are you sure you want to delete the work package"
```
Comment (Kharonus) [inline config/locales/js-en.yml:1140]: minor: ist there a `?` missing?

File: config/locales/js-en.yml [hunk 12]
```diff
@@ -1274,11 +1278,12 @@ en:
...
       message:
         label: 'Invitation message'
-        description: 'We will send an email to %{principal}, to which you can add a personal message here. An explanation for the invitation could be useful, or perhaps a bit of information regarding the project to help them get started.'
-        next_button: 'Review invitation'
+        description: 'We will send an email to the user, to which you can add a personal message here. An explanation for the invitation could be useful, or prehaps a bit of information regarding the rpoject to help them get started.'
+        next_button: 'Next'
 
       summary:
         next_button: 'Send invitation'
+        cancel_button: 'Cancel'
```
Comment (Kharonus) [inline config/locales/js-en.yml]: why having multiple `cancel` locale keys? we have the `js.button_cancel` key, which IMHO can be used for all of those above.

File: frontend/src/app/features/invite-user-modal/project-selection/project-selection.component.html [hunk 13]
```diff
@@ -62,9 +62,17 @@
...
-  <div class="spot-modal--footer spot-action-bar">
+  <div class="spot-action-bar">
     <div class="spot-action-bar--right">
+      <button
+      type="button"
+      class="button button_no-margin spot-action-bar--action"
+      (click)="close.emit()"
+    >
+    {{ text.cancelButton }}
+  </button>
```
Comment (Kharonus) [inline frontend/src/app/features/invite-user-modal/project-selection/project-selection.component.html]: minor: indention is off

**Review by b12f — APPROVED** [review]
LGTM, awesome work

File: frontend/src/app/shared/components/modals/save-modal/save-query.modal.html [hunk 14]
```diff
@@ -1,16 +1,16 @@
 <form
-  class="spot-modal wp-table--configuration-modal loading-indicator--location"
+  class="spot-modal loading-indicator--location"
   data-indicator-name="modal"
   name="modalSaveForm"
   (submit)="saveQueryAs($event)"
 >
-  <div class="spot-modal--header">{{text.save_as}}</div>
-
+  <h1 class="spot-modal--header">{{text.save_as}}</h1>
```
Comment (Kharonus) [inline frontend/src/app/shared/components/modals/save-modal/save-query.modal.html:7]: usually the header doesn't need to be `h1`. Why do we use it here instead of plain `div`?

File: frontend/src/app/spot/styles/sass/components/modal.sass [hunk 15]
```diff
@@ -38,7 +38,10 @@
     position: relative
     align-items: center
 
-    .spot-icon
+    + .spot-divider
+      margin-top: $spot-spacing-1
```
Comment (Kharonus) [inline frontend/src/app/spot/styles/sass/components/modal.sass:42]: this should not be needed? the top margin of the divider is defined in `divider.sass`

**Review by Kharonus — CHANGES_REQUESTED** [review]
I found some issues, which all should be easy fixes. Let me know, if there are any questions.

File: frontend/src/app/shared/components/modals/save-modal/save-query.modal.html (hunk 14 above)
Comment (b12f) [reply frontend/src/app/shared/components/modals/save-modal/save-query.modal.html:7]: Was my idea; it is semantically correct HTML. The advantage we have with the typography spot classes is that you can write HTML that is semantically correct (this is the main heading of the modal, thus `h1`) but is easily made to look correct in context

File: frontend/src/app/spot/styles/sass/components/modal.sass (hunk 15 above)
Comment (b12f) [reply frontend/src/app/spot/styles/sass/components/modal.sass:42]: We [removed this](https://github.com/opf/openproject/pull/11246/files#diff-64ce798fd9b6656777121aa833252cfe12e3d84a026c30eb47461ad38f88cc85L2) because it breaks the BEM assumption that blocks never have margins.

Instead we added them to the context right here (spot-modal) and [in spot-container](https://github.com/opf/openproject/pull/11246/files#diff-2246b4e29d74b4ce0bee934476e29eaed67eb2c59c08bdec4e8a1ce4e2f527d5R13).

Comment (b12f) [reply frontend/src/app/spot/styles/sass/components/modal.sass:42]: I actually think `spot-divider` should maybe always be an `hr` instead of a `div`, but that's a quick change we can make any time.

File: config/locales/js-en.yml (hunk 11 above)
Comment (bsatarnejad) [reply config/locales/js-en.yml:1140]: No this text will be followed by the WP subject and ID, and then will be a question mark in html.

File: frontend/src/app/shared/components/modals/save-modal/save-query.modal.html (hunk 14 above)
Comment (Kharonus) [reply frontend/src/app/shared/components/modals/save-modal/save-query.modal.html:7]: got it, no change needed then. I just feared, this might drag some unwanted margins in.

File: config/locales/js-en.yml (hunk 11 above)
Comment (Kharonus) [reply config/locales/js-en.yml:1140]: great

**Review by Kharonus — APPROVED** [review]
LGTM
